libsolidity/formal/SolverInterface.h

```diff
@@ -65,16 +65,20 @@ struct FunctionSort: public Sort
 {
 	FunctionSort(std::vector<SortPointer> _domain, SortPointer _codomain):
 		Sort(Kind::Function), domain(std::move(_domain)), codomain(std::move(_codomain)) {}
-	bool operator==(FunctionSort const& _other) const
+	bool operator==(Sort const& _other) const override
 	{
+		if (!Sort::operator==(_other))
+			return false;
+		auto _otherFunction = dynamic_cast<FunctionSort const*>(&_other);
+		solAssert(_otherFunction, "");
 		if (!std::equal(
```

Comment:
You need to check the size to be equal. I would recommend rather using a boost function that does this properly.

Reply:
I'd rather check it explicitly if that's also fine. Updated.